test/test_torch.py

```diff
@@ -4005,11 +4005,16 @@ def test_serialization(self):
         b += [(t1.storage(), t1.storage(), t2.storage())]
         b += [a[0].storage()[0:2]]
         for use_name in (False, True):
-            with tempfile.NamedTemporaryFile() as f:
+            delete = sys.platform != 'win32' or not use_name
+            with tempfile.NamedTemporaryFile(delete=delete) as f:
                 handle = f if not use_name else f.name
                 torch.save(b, handle)
                 f.seek(0)
                 c = torch.load(handle)
+                if not delete:
+                    del handle
+                    f.close()
+                    os.remove(f.name)
```

Comment:
So I still don't understand why is this necessary. The Python issue you linked mentions that if you want to re-open the same file for reading, then you have to close it first. However in here, none of these two things happen, and you do `os.remove()` at the end anyway (which seems like exactly what `delete=True` would do)

Reply:
I found the reason to that. It's the locking problem that blocks the file from opening again. According to the [post](https://stackoverflow.com/questions/546504/how-do-i-make-windows-file-locking-more-like-unix-file-locking), I need to append a flag in THAllocator.c